utility/sourcemap_accessor.go

```diff
@@ -0,0 +1,180 @@
+package utility
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/go-sourcemap/sourcemap"
+	lru "github.com/hashicorp/golang-lru"
+
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/outputs/elasticsearch"
+)
+
+type SmapAccessor interface {
+	Fetch(smapId SmapID) (*sourcemap.Consumer, error)
+}
+
+type Enum string
+
+const (
+	InitError   Enum = "InitError"
+	AccessError Enum = "AccessError"
+	MapError    Enum = "MapError"
+	ParseError  Enum = "ParseError"
+)
+
+type SmapError struct {
+	Msg  string
+	Kind Enum
+}
+
+func (e SmapError) Error() string {
+	return fmt.Sprintf("%s", e.Msg)
+}
+
+type SmapID struct {
+	ServiceName    string
+	ServiceVersion string
+	Path           string
+}
+
+type SourcemapAccessor struct {
+	esClients []elasticsearch.Client
+	smapCache *lru.Cache
+}
+
+func NewSourcemapAccessor(esClients []elasticsearch.Client, cacheSize int) (*SourcemapAccessor, error) {
+	if esClients == nil || len(esClients) == 0 {
+		err := SmapError{Msg: "Sourcemap ESClient cannot be initialized.", Kind: InitError}
+		logp.Err(err.Error())
+		return nil, err
+	}
+
+	smapCache, err := lru.New(cacheSize)
```

Comment:
is there a way to get the number of used or available slots in the lru cache? i think this would be a useful number to have for debugging potential memory issues. We could use the beats metrics to keep track of it.